site/pages/api/type-of-consequence-template.test.ts

```diff
@@ -0,0 +1,211 @@
+
+    it("should redirect to services consequence page when 'Services' is selected", () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: { ...disruptionData, consequenceType: "services" },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        addConsequenceTemplate(req, res);
+
+        expect(writeHeadMock).toBeCalledWith(302, { Location: `/create-consequence-services/${disruptionId}/0` });
+    });
+
+    it("should redirect back to add consequence page (/type-of-consequence) when no inputs are passed", () => {
+        const errors: ErrorInfo[] = [{ errorMessage: "Select a consequence type", id: "consequenceType" }];
+
+        const { req, res } = getMockRequestAndResponse({
+            body: { ...disruptionData, consequenceType: "" },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        addConsequenceTemplate(req, res);
+
+        expect(setCookieOnResponseObject).toHaveBeenCalledTimes(1);
+        expect(setCookieOnResponseObject).toHaveBeenNthCalledWith(
+            1,
+            COOKIES_CONSEQUENCE_TYPE_ERRORS,
+            JSON.stringify({ inputs: req.body, errors }),
+            res,
+        );
+
+        expect(writeHeadMock).toBeCalledWith(302, { Location: `${TYPE_OF_CONSEQUENCE_PAGE_PATH}/${disruptionId}/0` });
+    });
+
+    it("should redirect back to add consequence page (/type-of-consequence) when incorrect values are passed", () => {
+        const errors: ErrorInfo[] = [{ errorMessage: "Select a consequence type", id: "consequenceType" }];
+
+        const { req, res } = getMockRequestAndResponse({
+            body: { ...disruptionData, consequenceType: "incorrect type" },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        addConsequenceTemplate(req, res);
+
+        expect(setCookieOnResponseObject).toHaveBeenCalledTimes(1);
+        expect(setCookieOnResponseObject).toHaveBeenNthCalledWith(
+            1,
+            COOKIES_CONSEQUENCE_TYPE_ERRORS,
+            JSON.stringify({ inputs: req.body, errors }),
+            res,
+        );
+
+        expect(writeHeadMock).toBeCalledWith(302, { Location: `${TYPE_OF_CONSEQUENCE_PAGE_PATH}/${disruptionId}/0` });
+    });
+
+    it("should redirect to operator consequence page when 'Operator wide' selected and with appropriate query params when a new disruption is created from template", () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: disruptionData,
+            requestHeaders: {
+                referer: refererPath,
+            },
+            mockWriteHeadFn: writeHeadMock,
+        });
+
+        addConsequenceTemplate(req, res);
+
+        expect(writeHeadMock).toBeCalledWith(302, {
+            Location: `/create-consequence-operator/${disruptionId}/0`,
+        });
+    });
+
+    it("should redirect to operator consequence page when 'Network wide' selected and with appropriate query params when a new disruption is created from template", () => {
```

Comment:
Not sure if this test is required because we don't expect any query params when in the template flow

Reply:
removed